applications/dashboard/controllers/api/AbstractApiController.php

```diff
@@ -44,6 +44,47 @@ public function formatField(array &$row, $field, $format) {
     }
 
     /**
+     * Determine which fields should be expanded, using a request and a field map.
+     *
+     * @param array $data An array representing request data.
+     * @param array $map An array of short-to-full field names (e.g. insertUser => InsertUserID).
+     * @param string $field The name of the field where the expand fields can be found.
+     * @return array
+     */
+    protected function getExpandFields(array $data, array $map, $field = 'expand') {
+        $result = [];
+        if (array_key_exists($field, $data)) {
+            $expand = $data[$field];
+            foreach ($map as $short => $full) {
+                if (in_array($short, $expand)) {
+                    $result[] = $full;
+                }
+            }
+        }
+        return $result;
+    }
+
+    /**
+     * Get a simple schema for nesting as an "expand" parameter.
+     *
+     * @param array $fields Valid values for the expand parameter.
+     * @return Schema
+     */
+    protected function getExpandFragment(array $fields) {
+        $result = $this->schema([
```

Comment:
This should not be a model of it's own. Use `Schema::parse();`
Or if we want to expose it as it's own model the enum part should be set to '...' on the first declaration and set properly before being returned otherwise the first call to it will set the items to something that will probably not be relevant for many endpoints.